spec/datetimeformat.html

```diff
@@ -94,9 +94,18 @@ <h1>
           1. Set _timeZone_ to DefaultTimeZone().
         1. Else,
           1. Set _timeZone_ to ? ToString(_timeZone_).
-          1. If the result of IsValidTimeZoneName(_timeZone_) is *false*, then
-            1. Throw a *RangeError* exception.
+        1. If IsTimeZoneOffsetString(_timeZone_) is *true*, then
+          1. Let _parseResult_ be ParseText(StringToCodePoints(_timeZone_), |UTCOffset|).
+          1. Assert: _parseResult_ is a Parse Node.
+          1. If _parseResult_ contains more than one |MinuteSecond| Parse Node, throw a *RangeError* exception.
```

Comment:
I am not sure how does "contains more than one |MinuteSecond| Parse Node" work here.
How is the "contains" operation defined for Parse Node. I cannot find a concrete definition. 

Look at the UTCOffset grammar in https://tc39.es/ecma262/#sec-time-zone-offset-string-format
```
UTCOffset :::
TemporalSign Hour
TemporalSign Hour HourSubcomponents[+Extended]
TemporalSign Hour HourSubcomponents[~Extended]
```

So if the concept of "contains" is shallow, then it would be impossible for UTCOffset to contain any MinuteSecond.
since 1 or 2 MinuteSecond could be contain inside the HourSubcomponents that contained by UTCOffset. However, the require a "deep containment" interpretation and I am not sure that is what ECMA262 specify. (since I cannot find a clear definition of what  "contains" mean for Parse Node.

Reply:
@FrankYFTang This use of "contains" is prose of the sort that is also currently in Temporal [ParseTimeZoneIdentifier](https://tc39.es/proposal-temporal/#sec-parsetimezoneidentifier)—as suggested by the lack of automatic hyperlinking, there is no concrete definition. However, as I thought would be clear from context in both documents, it is intended to be understood as entailing deep inspection of the tree of Parse Nodes. Can you suggest a change here that would make that more clear?

I could define a precise syntax-directed operation, but **really** don't want to do that because it would apply cross-document to the [ECMA-262 |UTCOffset| nonterminal](https://tc39.es/ecma262/multipage/numbers-and-dates.html#prod-UTCOffset) that will be _replaced_ by [Temporal |TimeZoneUTCOffsetName|](https://tc39.es/proposal-temporal/#prod-TimeZoneUTCOffsetName) (creating a situation where this part of ECMA-402 would be incoherent until fixed).

It's also worth noting that this explicit check for multiple |MinuteSecond| Parse Nodes will itself be removed when Temporal lands, along with many of the other steps here that are subsumed by [ParseTimeZoneIdentifier](https://tc39.es/proposal-temporal/#sec-parsetimezoneidentifier):
```patch
-        1. If IsTimeZoneOffsetString(_timeZone_) is *true*, then
+        1. Let _timeZoneRecord_ be ? ParseTimeZoneIdentifier(_timeZone_).
+        1. If _timeZoneRecord_.[[Name]] is ~empty~, then
-          1. Let _parseResult_ be ParseText(StringToCodePoints(_timeZone_), |UTCOffset|).
-          1. Assert: _parseResult_ is a Parse Node.
-          1. If _parseResult_ contains more than one |MinuteSecond| Parse Node, throw a *RangeError* exception.
-          1. Let _offsetNanoseconds_ be ParseTimeZoneOffsetString(_timeZone_).
-          1. Let _offsetMinutes_ be _offsetNanoseconds_ / (6 × 10<sup>10</sup>).
-          1. Assert: _offsetMinutes_ is an integer.
+          1. Let _offsetMinutes_ be _timeZoneRecord_.[[OffsetMinutes]].
           1. Set _timeZone_ to FormatOffsetTimeZoneIdentifier(_offsetMinutes_).
-        1. Else if IsValidTimeZoneName(_timeZone_) is *true*, then
+        1. Else if IsValidTimeZoneName(_timeZoneRecord_.[[Name]]) is *true*, then
           1. Set _timeZone_ to CanonicalizeTimeZoneName(_timeZoneRecord_.[[Name]]).
         1. Else,
           1. Throw a *RangeError* exception.
```